### PR TITLE
Nez 176

### PR DESCRIPTION
**Pull request by sakshi2894**
Ability to set current state of migration.
Support downgrade in migrations.
Upgrade/Downgrade a specific revision

### CODE REVIEW

File: migrations/migrations.py [hunk 1]
```diff
@@ -46,12 +46,35 @@ def list(self):
...
         return revisions
 
-    def upgrade(self):
+    def upgrade(self, version=None, current=-1):
+        if version is not None:
+            version = int(version)
+            self.map[version].module.upgrade(self.context)
+            return
+
+        current = int(current)
```
Comment (vrajat) [inline migrations/migrations.py]: Why did you move away from `self.current` ? Seems like `current` is serving the same purpose.

Comment (sakshi2894) [inline migrations/migrations.py]: was trying out something. Had to revert this before pushing - missed it.

File: migrations/migrations.py [hunk 2]
```diff
@@ -46,7 +46,14 @@ def list(self):
...
         return revisions
 
-    def upgrade(self):
+    def upgrade(self, version=None, current=-1):
+        if version is not None:
+            version = int(version)
+            self.map[version].module.upgrade(self.context)
+            return
+
+        self.current = int(current)
```
Comment (vrajat) [inline migrations/migrations.py]: `self.current` should be set in the constructor

**Comment by vrajat** [comment]
Need a test. Look at https://github.com/vrajat/python-migrations/blob/master/tests/test_migration.py#L23

**Comment by vrajat** [comment]
Needs a test. Good to go other than that
